### PR TITLE
Doc fixes & contributing guide

### PR DESCRIPTION
**Pull request by encukou**
Some documentation on how to contribute and maintain python-ldap.

Incidentally, this
Fixes: https://github.com/python-ldap/python-ldap/issues/50

### CODE REVIEW

File: Doc/contributing.rst [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+To run a subset of test environments, run for example::
+
+    (__venv__)$ tox -e py27,py36
+
+In addition to ``pyXY`` environments, we have extra environments
+for checking things independent of the Python version:
+
+* ``doc`` checks syntax and spelling of the documentation
+* ``coverage`` generates a test coverage report
```
Comment (tiran) [inline Doc/contributing.rst]: * ``coverage-report`` generates a test coverage report for Python code. It must be used as last test env, e.g. ``tox -e py27,py36,coverage-report``.

File: Doc/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+To run tests, install and run `tox`_::
+
+    (__venv__)$ python -m pip install tox
+    (__venv__)$ tox
+
+This tries to run tests on all supported versions of Python.
+If you don't have some of them installed, you will get an
+``InterpreterNotFound`` error, which you can ignore::
+
+    ERROR:   py37: InterpreterNotFound: python3.7
```
Comment (tiran) [inline Doc/contributing.rst]: tox has ``skip_missing_interpreters``, https://tox.readthedocs.io/en/latest/config.html#confval-skip_missing_interpreters=BOOL Wanna add it?

Comment (encukou) [reply Doc/contributing.rst]: Yes! Thanks!

File: Doc/contributing.rst (hunk 1 above)
Comment (encukou) [reply Doc/contributing.rst]: Thanks for the catch!

**Pull request merged**